src/relax/ir/block_builder.cc

```diff
@@ -713,6 +713,9 @@ GlobalVar BlockBuilderNode::AddFunction(const BaseFunc& func, const String& func
   if (it == func_map_.end()) {
     context_mod_.CopyOnWrite();
     String func_name = name_table_->GetUniqueName(func_name_hint);
+    while (context_mod_->ContainGlobalVar(func_name)) {
+      func_name = name_table_->GetUniqueName(func_name_hint);
+    }
```

Comment:
This is a hack to find a correct name to the function which is to be inserted into the context IRModule.

The issue behind this hack is, recall that when creating a BlockBuilder, we can initialize the context IRModule with a given input IRModule.
https://github.com/mlc-ai/relax/blob/05914dbc186f96b9966e9450acfcc2a53022b327/src/relax/ir/block_builder.cc#L739-L743

However, the name table of the BlockBuilder is not initialized accordingly, which means even though we initialize the context IRModule, the name table is still empty.

This is actually not a very big deal if we only use the name table to create Vars, since variable names are allowed to duplicate. However, this is an issue when we are looking for a name for a function - since the name table is not initialized, the returning name (which is indeed unique from the name table side) is possible to duplicate with an existing function in the context IRModule. In this case, adding the new function to the IRModule will result in an error, since the `name_hint` of the input GlobalVar already exists in the GlobalVar mapping:

https://github.com/mlc-ai/relax/blob/05914dbc186f96b9966e9450acfcc2a53022b327/src/ir/module.cc#L222-L227

---

A possible solution to this issue is to initialize the name table accordingly when creating BlockBuilder with an initializing IRModule, by scanning each function in the IRModule and adding the appearing names into the name table.

cc @YuchenJin

Reply:
Thanks for noting this issue! Adding a helper function to NameTable by taking an IRModule and fill the hashmap sounds like a good plan.